ros2topic/ros2topic/verb/echo.py

```diff
@@ -105,33 +107,8 @@ def subscriber(
     qos_profile: QoSProfile
 ) -> Optional[str]:
     """Initialize a node with a single subscription and spin."""
-    if message_type is None:
-        topic_names_and_types = get_topic_names_and_types(node=node, include_hidden_topics=True)
-        try:
-            expanded_name = expand_topic_name(topic_name, node.get_name(), node.get_namespace())
-        except ValueError as e:
-            raise RuntimeError(e)
-        try:
-            validate_full_topic_name(expanded_name)
-        except rclpy.exceptions.InvalidTopicNameException as e:
-            raise RuntimeError(e)
-        for n, t in topic_names_and_types:
-            if n == expanded_name:
-                if len(t) > 1:
-                    raise RuntimeError(
-                        "Cannot echo topic '%s', as it contains more than one type: [%s]" %
-                        (topic_name, ', '.join(t))
-                    )
-                message_type = t[0]
-                break
-        else:
-            raise RuntimeError(
```

Comment:
I think we need to add back a similar guard to avoid passing `None` to `rclpy.Node.create_subscription`.
This introduced a bug reported in https://github.com/ros2/ros2cli/issues/519

Reply:
See fix in https://github.com/ros2/ros2cli/pull/520